src/apps/factory/management/commands/create_competition.py

```diff
@@ -2,21 +2,391 @@
+from competitions.models import Competition, Phase, CompetitionParticipant
+from profiles.models import User as CodalabUser
+
+from tqdm import tqdm
+
+from faker import Faker
+fake = Faker()
+
+
+def _create_fake_user():
+    try:
+        # Usernames will be in the format <some_username><some_number> IE: shieldshannah53
+        temp_email = fake.safe_email()
+        temp_email = "{0}{1}{2}".format(temp_email.split('@')[0] + "@", random.randint(1,9999), temp_email.split('@')[1])
+        new_user = CodalabUser.objects.create(
+            username="{0}{1}".format(fake.user_name(), random.randint(1,9999)),
+            name=fake.name(),
+            email=temp_email,
+        )
+        # print(colored("Successfully created new user!", 'green'))
+        return new_user
+    except:
+        print(colored("Could not create fake user! Operation failed!", 'red'))
+        raise ObjectDoesNotExist("Could not create fake user! Operation failed!")
+
+
+def _create_fake_participant(competition):
+    try:
+        new_user = _create_fake_user()
+        new_part = CompetitionParticipant.objects.create(
+            competition=competition,
+            user=new_user
+        )
+        # print(colored("Successfully created new participant!", 'green'))
+        return new_part
+    except:
+        print(colored("Could not create fake participant! Operation failed!", 'red'))
+        raise ObjectDoesNotExist("Could not create fake participant! Operation failed!")
+
+
+def _stringdate_to_datetime(str_date):
+    print("STRING DATE IS: {}".format(str_date))
+    # DD-MM-YYYY:HH-MM-SS'
+    temp_date = str_date.split(':')[0]
+    temp_time = str_date.split(':')[1]
+
+    temp_day = int(temp_date.split('-')[0])
+    temp_month = int(temp_date.split('-')[1])
+    temp_year = int(temp_date.split('-')[2])
+
+    temp_hour = int(temp_time.split('-')[0])
+    temp_minute = int(temp_time.split('-')[1])
+    temp_second = int(temp_time.split('-')[2])
+
+    temp_new_date = datetime.datetime(
+        day=temp_day,
+        month=temp_month,
+        year=temp_year,
+        hour=temp_hour,
+        minute=temp_minute,
+        second=temp_second
+    )
+    return temp_new_date
+
+
+def _check_data_key(options, options_key, data, data_key, default_value, is_date=False):
+    if not options or not options_key:
+        data[data_key] = default_value
+    elif options.get(options_key):
+        if is_date:
+            data[data_key] = _stringdate_to_datetime(options[options_key])
+        else:
+            data[data_key] = options[options_key]
+    elif not data.get(data_key):
+        data[data_key] = default_value
+
+
+def _create_competition(full_details, data):
+    if full_details:
+        # should_have_prize = random.randint(1, 4) == 4
+        # if not should_have_prize:
+        #     data['prize'] = None
+        fake_url = 'competitions/{0}/'.format(random.randint(0, 9999))
+        temp_competition = Competition.objects.create(
```

Comment:
It'd be ideal if we could access this from anywhere. We should really really really be using model mommy, factory boy, or what-have-you so we can also use this competition creation stuff in tests.

https://factoryboy.readthedocs.io/en/latest/

```python
class RandomUserFactory(factory.Factory):
    class Meta:
        model = models.User

    first_name = factory.Faker('first_name')
    last_name = factory.Faker('last_name')
```